R/dt_options.R

```diff
@@ -49,6 +49,24 @@ dt_options_get_value <- function(data, option) {
   dt_options$value[[which(dt_options$parameter == option)]]
 }
 
+# Get a list of option values
+dt_options_get_values <- function(data) {
+  dt_options <- dt_options_get(data = data)[c("parameter", "value")]
+  # Similar to tibble::deframe
+  res <- vctrs::vec_set_names(dt_options$value, dt_options$parameter)
+  class(res) <- c("gt_option", class(res))
+  res
+}
+
+#' @export
+`$.gt_option` <- function(x, name) {
+  out <- .subset2(x, name)
+  if (is.null(out)) {
+    cli::cli_abort("Can't find option {.val {name}}.")
+  }
+  out
+}
```

Comment:
This is to avoid a typo when developing `dt_options_get_values(data)$olumn_label` errors instead of returning `NULL`, similar to the method that tibble defines for tables. I have been bitten while developing, so I thought it was better to catch the error on the spot for easier debugging

```r
exibble$num2

#> Warning: Unknown or uninitalized column
mtcars$nonexistent
#> NULL
```

Reply:
I like it!